src/validation.cpp

```diff
@@ -1672,6 +1675,23 @@ void ThreadScriptCheck(int worker_num) {
     scriptcheckqueue.Thread();
 }
 
+bool RunCheckInputsMaybeParallel(const CTransaction& tx, CValidationState &state, const CCoinsViewCache &inputs, unsigned int flags, bool cacheSigStore, bool cacheFullScriptStore, PrecomputedTransactionData& txdata, bool single_threaded_validation) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
+{
+    uint256 hash_cache_entry;
+    CCheckQueueControl<CScriptCheck> control((nScriptCheckThreads && !single_threaded_validation) ? &scriptcheckqueue : nullptr);
+    std::vector<CScriptCheck> vChecks;
+    if (!CheckInputs(tx, state, inputs, true, flags, cacheSigStore, cacheFullScriptStore, txdata, hash_cache_entry, (nScriptCheckThreads && !single_threaded_validation) ? &vChecks : nullptr)) {
+        return false;
+    }
+    control.Add(vChecks);
+    if (!control.Wait()) {
+        return state.Invalid(ValidationInvalidReason::CONSENSUS, false, REJECT_INVALID, "script-verify-flag-failed");
+    }
+    if (nScriptCheckThreads && !single_threaded_validation && cacheFullScriptStore) {
```

Comment:
nit: since `(nScriptCheckThreads && !single_threaded_validation)` is used several times throughout this function, maybe declare it at the beginning of the function? (e.g. `bool use_multiple_threads = nScriptCheckThreads && !single_threaded_validation`).

Reply:
Done